app/javascript/src/LoginPage/loginForms/SignupForm.jsx

```diff
@@ -53,68 +74,75 @@ const InputFormGroup = (props) => {
 }
 
 class SignupForm extends Component<Props, State> {
-  constructor (props: Props) {
-    super(props)
-    this.state = {
-      username: this.props.user.username,
-      emailAddress: this.props.user.email,
-      firstname: this.props.user.firstname,
-      lastname: this.props.user.lastname,
-      password: '',
-      passwordConfirmation: '',
-      isFilledUsername: undefined,
-      isFilledEmailAddress: undefined,
-      isFilledFirstname: true,
-      isFilledLastname: true,
-      isFilledPassword: undefined,
-      isFilledPasswordConfirmation: undefined
+  state = {
+    username: this.props.user.username,
+    email: this.props.user.email,
+    firstName: this.props.user.firstname,
+    lastName: this.props.user.lastname,
+    password: '',
+    passwordConfirmation: '',
+    validation: {
+      username: undefined,
+      email: undefined,
+      firstName: true,
+      lastName: true,
+      password: undefined,
+      passwordConfirmation: undefined
     }
   }
 
-  handleTextInputUsername = (username: string) =>
-    this.setState({ username, isFilledUsername: username !== '' })
-
-  handleTextInputEmail = (emailAddress: string) =>
-    this.setState({ emailAddress, isFilledEmailAddress: emailAddress !== '' })
-
-  handleTextInputFirstname = (firstname: string) => this.setState({ firstname })
-
-  handleTextInputLastname = (lastname: string) => this.setState({ lastname })
+  handleInputChange = (value: string, event: SyntheticEvent<HTMLInputElement>) => {
+    const isValid = event.currentTarget.required ? validateSingleField(event) : true
+    const validation = {
+      ...this.state.validation,
+      [namesToStateKeys[event.currentTarget.name]]: isValid
+    }
 
-  handleTextInputPassword = (password: string) =>
     this.setState({
-      password,
-      isFilledPassword: password !== '',
-      isFilledPasswordConfirmation: this.state.passwordConfirmation === password
+      [namesToStateKeys[event.currentTarget.name]]: value,
+      validation
     })
+  }
 
-  handleTextInputPasswordConfirmation = (passwordConfirmation: string) =>
-    this.setState({
-      passwordConfirmation,
-      isFilledPasswordConfirmation: passwordConfirmation !== '' && passwordConfirmation === this.state.password
-    })
+  renameValidation = (namesToStateKeys: NamesToKeys, invalidFields: InvalidFields) => Object.keys(invalidFields)
```

Comment:
What is this function for? What does it mean?

Reply:
This is needed because validate.js uses the name attribute to access the inputs.
In the other pages names are in the format `password`, or `email`, but in this page the name attrs are in the format `user[email]` , `user[password]`, etc.
For this reason I'm keeping the `namesToStateKeys` object only in this form.
That attributes comes from rails, we can't change them in the scope of this task